config/static/nodes/cri-o/install-custom-cri-o.yaml.envsubst

```diff
@@ -0,0 +1,30 @@
+---
+# https://github.com/openshift/machine-config-operator/tree/master/docs
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: 99-install-custom-cri-o-${ROLE}
+  labels:
+    machineconfiguration.openshift.io/role: ${ROLE}
+spec:
+  config:
+    ignition:
+      version: "3.2.0"
+    systemd:
+      units:
+      - name: "install-cri-o.service"
+        enabled: true
+        # For getting the logs run:
+        # $ oc adm node-logs node/NODE -u install-cri-o
+        contents: |
+          [Unit]
+          Description=Install custom cri-o
+          # https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/
+          After=network-online.target
```

Comment:
This unit does not work for me on OCP 4.10.  Logs:
```
% oc adm node-logs node/ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx -u rpm-overrides.service                                                                    
-- Logs begin at Thu 2021-12-02 02:52:45 UTC, end at Thu 2021-12-02 05:20:26 UTC. --
-- No entries --
-- Logs begin at Thu 2021-12-02 02:52:45 UTC, end at Thu 2021-12-02 05:20:26 UTC. --
Dec 02 05:00:24.321949 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx systemd[1]: Starting Install RPM overrides...
Dec 02 05:00:24.339044 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: package runc-1.0.2-992.rhaos4.10.el8.x86_64 is not installed
Dec 02 05:00:25.177411 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Downloading 'https://ftweedal.fedorapeople.org/runc-1.0.2-992.rhaos4.10.el8.x86_64.rpm'... done!
Dec 02 05:00:29.025484 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Checking out tree d95b9a4...done
Dec 02 05:00:29.089007 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: No enabled rpm-md repositories.
Dec 02 05:00:29.203517 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Importing rpm-md...done
Dec 02 05:00:29.212301 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Resolving dependencies...done
Dec 02 05:00:29.214332 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Applying 1 override
Dec 02 05:00:29.306524 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Processing packages...done
Dec 02 05:00:29.309166 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Running pre scripts...done
Dec 02 05:00:29.310469 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Running post scripts...done
Dec 02 05:00:30.232707 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Running posttrans scripts...done
Dec 02 05:00:30.420456 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Writing rpmdb...done
Dec 02 05:00:32.387732 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Writing OSTree commit...done
Dec 02 05:00:35.136481 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: Staging deployment...done
Dec 02 05:00:35.137815 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx sh[1355]: error: Bus owner changed, aborting. This likely means the daemon crashed; check logs with `journalctl -xe`.
Dec 02 05:00:35.141900 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx systemd[1]: rpm-overrides.service: Main process exited, code=exited, status=1/FAILURE
Dec 02 05:00:35.142056 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx systemd[1]: rpm-overrides.service: Failed with result 'exit-code'.
Dec 02 05:00:35.142501 ci-ln-m7jtknk-72292-9xv9c-worker-c-v74jx systemd[1]: Failed to start Install RPM overrides.
```

Perhaps it is being executed too early in boot sequence?  Perhaps it needs to run `After=rpm-ostreed.service`?

Reply:
Thanks @frasertweedale ; I tested on OCP 4.9.5; I am installing an OCP 4.10 for fixing this.

Reply:
I have tried in OCP 4.9.9 and it was successful too; the output I get is the below:

```raw
-- Logs begin at Fri 2021-11-26 04:40:34 UTC, end at Thu 2021-12-02 21:08:58 UTC. --
Dec 02 21:07:01 crc-x4qnm-master-0 bash[1663]: package runc-1.0.0-990.rhaos4.8.gitcd80260.el8.x86_64 is not installed
Dec 02 21:07:01 crc-x4qnm-master-0 systemd[1]: Starting Install custom runc...
Dec 02 21:07:05 crc-x4qnm-master-0 bash[1663]: Downloading 'https://ftweedal.fedorapeople.org/runc-1.0.0-990.rhaos4.8.gitcd80260.el8.x86_64.rpm'... done!
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Checking out tree b690184...done
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: No enabled rpm-md repositories.
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Importing rpm-md...done
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Resolving dependencies...done
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Applying 1 override and 5 overlays
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Processing packages...done
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Running pre scripts...done
Dec 02 21:07:07 crc-x4qnm-master-0 bash[1663]: Running post scripts...done
Dec 02 21:07:08 crc-x4qnm-master-0 bash[1663]: Running posttrans scripts...done
Dec 02 21:07:08 crc-x4qnm-master-0 bash[1663]: Writing rpmdb...done
Dec 02 21:07:09 crc-x4qnm-master-0 bash[1663]: Writing OSTree commit...done
Dec 02 21:07:10 crc-x4qnm-master-0 bash[1663]: Staging deployment...done
Dec 02 21:07:11 crc-x4qnm-master-0 systemd[1]: install-runc.service: Succeeded.
Dec 02 21:07:11 crc-x4qnm-master-0 systemd[1]: Stopped Install custom runc.
Dec 02 21:07:11 crc-x4qnm-master-0 systemd[1]: install-runc.service: Consumed 161ms CPU time
-- Reboot --
Dec 02 21:08:00 crc-x4qnm-master-0 bash[1648]: runc-1.0.0-990.rhaos4.8.gitcd80260.el8.x86_64
Dec 02 21:08:00 crc-x4qnm-master-0 systemd[1]: Starting Install custom runc...
Dec 02 21:08:00 crc-x4qnm-master-0 systemd[1]: install-runc.service: Succeeded.
Dec 02 21:08:00 crc-x4qnm-master-0 systemd[1]: Started Install custom runc.
Dec 02 21:08:00 crc-x4qnm-master-0 systemd[1]: install-runc.service: Consumed 15ms CPU time
```
Comparing with the output that you get, I can see an error message at the end; the message indicate the daemon crashed; I want to try in OCP 4.10 to dig more on this and replay it. I am having some situations to get the cluster ready.

By the way, your traces inspired me for getting a more generalized overlay for installing the rpm packages, thank you; I have updated the PR so that now we have only one overlay that can be customized for the POOL and the list of packages. I have tested in CodeReadyContainers so far.

Reply:
Unfortunately, I get the exact same failure on OCP 4.9 (nightly) via cluster-bot.  Must be something different about how nodes are managed in CRC.

Recommend you install cluster(s) on PSI or via cluster-bot for testing.

Reply:
Thanks, I replayed the situation in OCP 4.9.9 in PSI.

Reply:
https://www.freedesktop.org/software/systemd/man/systemd.special.html

> A target that should be used as synchronization point for all host/network name service lookups.

Reply:
thanks @tiran for the comments and the link to the documentation; updating the systemd unit generated.